### ISSUE TITLE
Can the newly upgraded version of ANNEVO (v2.2.3) use models pretrained with v2.2.2?

### ISSUE THREAD

**Issue by Wanchengshan** [xjtu-omics/ANNEVO#25]
Does this new upgrade of ANNEVO (v2.2.3) include any upgrades related to the pretrained models? I just obtained models pretrained with v2.2.2. Is it still necessary to re-pretrain using v2.2.3? Would you recommend that I use the v2.2.3 software to make predictions based on models pretrained with v2.2.2? By the way, I pretrained models for birds, and plants are not involved for now.

**Comment by PyZhang-Bio** [comment]
The new models provided in v2.2.3 differ from the previous approach, so at the current stage the latest training strategy and scripts may not yet be available.

For now, for avian species, you can still continue using the v2.2.2 model. However, we plan to release a major update in the near future. In particular, for animals, we will also adopt a boundary-aware strategy, together with longer-context training and context-extension approaches. These changes should be especially effective for long genes and help prevent issues such as erroneous gene fragmentation.

Thank you for your interest and support.

Best regards,
Pengyu

**Comment by Wanchengshan** [comment]
Great to hear that a major update is on the way! Could you give a rough estimate of how much improvement the next version might bring? I'm really looking forward to it. By the way, approximately when do you expect the new version to be released? If it's going to take too long, I may have to settle for the current version in the meantime.

**Comment by PyZhang-Bio** [comment]
I cannot give a very definite timeline at the moment, but it should not take more than one or two months. Since we do not plan to publish a separate paper for the new version of the sequence model, but instead release it directly for public use, this should greatly speed up the update process.

We expect to make a relatively major update in the next version, v2.3, which will also include some minor adjustments to the model architecture. Therefore, we will first release several major branches, and birds are expected to be one of them. We will then continue to gradually update the remaining branches afterward.

**Comment by Wanchengshan** [comment]
I used V2.2.2 to fine-tune a bird model on a non-mammalian vertebrate model, performing 30 epochs of training. The final losses are as follows:

total_loss_CE: 0.046722281656011284
total loss_dice_CDS: 0.07274132348163245
total loss dice intron: 0.0350651428495059
total loss balance: 0.0013667502602334855

The trained .pt file was successfully generated, and its file size is consistent with the example models. No errors occurred during training.

However, unfortunately, when I use this newly trained model for prediction — whether with V2.2.2 or V2.2.3 — I am unable to successfully predict any genes. It only generates a GFF file, but no genes are predicted for any of the chromosomes. Note that the prediction runtime is comparable to a normal prediction. When I use the example models you provided, gene prediction works successfully.

Could the author please help me troubleshoot the possible cause?

nohup python ANNEVO-2.2.2/fine_tune.py \
    --model_path ./ANNEVO-2.2.2/ANNEVO_model/ANNEVO_Vertebrate_other.pt \
    --model_save_path ./ANNEVO_Strigiformes_finetuned.pt \
    --h5_path ./ANNEVO-2.2.2/fine_tune_data/ \
    --batch_size 56 \
    --epoch 30 \
    --learning_rate 1e-5 > fine_tune_restart.log 2>&1 &

python ./ANNEVO-main/annotation.py \
  --genome ../ref_genomes/A.cuni/GCF_003259725.1_athCun1_genomic.fna\
  --model_path ANNEVO-main/ANNEVO_model/ANNEVO_Strigiformes_finetuned.pt \
  --output ./output/A.cuni_ANNEVO_Strigiformes_annotation.gff \
  --batch_size 4 \
  --threads 54

head -n 20 A.cuni_ANNEVO_Strigiformes_annotation.gff
# This output was generated with ANNEVO v2.2.2.
# ANNEVO is an ab initio gene annotation tool written by YeLab.
# Citation: Zhang, P., Xu, T., Wang, S. et al. Highly accurate ab initio gene annotation with ANNEVO. Nat Methods (2026). https://doi.org/10.1038/s41592-026-03036-7
#
# ----- prediction on sequence number 1 (length = 714415, name = NW_020799094.1) -----
#
# Predicted genes for sequence number 1 on forward strands
# None
###
#
# Predicted genes for sequence number 1 on reverse strands
# None
###
#
# ----- prediction on sequence number 2 (length = 1040967, name = NW_020799093.1) -----
#
# Predicted genes for sequence number 2 on forward strands
# None
###
#

**Comment by PyZhang-Bio** [comment]
One possible explanation is that the model may have overfitted. I noticed that you fine-tuned it for 30 epochs, whereas in most cases I would expect 1–5 epochs to be sufficient. Therefore, although the loss value itself may still look reasonable, the model may have developed substantial generalization issues when applied to other species.

As a quick and simple test for comparison, I just fine-tuned the model starting from ANNEVO_Embryphyta and trained it for two epochs. We evaluated the re-annotated Arabidopsis genome using brassicales_odb12, and the BUSCO score was 99.2%. Please note that we did not fine-tune from the latest plant model, because the scripts for the new version have not yet been publicly released. Therefore, it is expected that this performance is not as good as that of the latest plant model.

I also noticed that you used a smaller learning rate and a larger batch size. The former makes each update extremely small, while the latter reduces the total number of optimization steps. I cannot say exactly how large the combined effect of these two factors is, but in general I think they could still have an impact.

In addition, we do not particularly recommend fine-tuning using data from only one or two species, as this can make the model very prone to overfitting. This is especially relevant for ANNEVO, which includes sparse MoE layers. Even though we selectively freeze part of the parameters, the model can still be relatively susceptible to overfitting during fine-tuning. This is consistent with observations from sparse expert models, for example: Zoph, B. et al. ST-MoE: Designing Stable and Transferable Sparse Expert Models.

**Comment by Wanchengshan** [comment]
Hello, Zhang.  Following your suggestions, I adjusted the fine-tuning strategy. I used 9 NCBI-annotated species for training, set the batch_size to 32, the epoch to 3, and the learning_rate to 0.0001. The final results after training are as follows:


total_loss_CE: 0.04702596458287987
total_loss_dice_CDS: 0.0728223143683802
total_loss_dice_intron: 0.03504265274944621
total_loss_balance: 0.0013706484959973909


However, unfortunately, even though the fine-tuned model file was successfully generated, it still failed to annotate the genomes — even the genomes used for training could not be successfully annotated.


Below are my training script and annotation script.

python ANNEVO-2.2.2/fine_tune.py     --model_path ./ANNEVO-2.2.2/ANNEVO_model/ANNEVO_Vertebrate_other.pt     --model_save_path ./ANNEVO_Strigiformes_finetuned2.pt     --h5_path ANNEVO-2.2.2/fine_tune_data/     --batch_size 32     --epoch 3  --learning_rate 0.0001

python ./ANNEVO-2.2.2/annotation.py --genome GCF_003259725.1_athCun1_genomic.fna --model_path ANNEVO_Strigiformes_finetuned2.pt --output A.cuni_ANNEVO_annotation.gff --batch_size 16 --threads 54

This output was generated with ANNEVO v2.2.2.
 ANNEVO is an ab initio gene annotation tool written by YeLab.
 Citation: Zhang, P., Xu, T., Wang, S. et al. Highly accurate ab initio gene annotation with ANNEVO. Nat Methods (2026). https://doi.org/10.1038/s41592-026-03036-7

 ----- prediction on sequence number 1 (length = 714415, name = NW_020799094.1) -----
 Predicted genes for sequence number 1 on forward strands
None


 Predicted genes for sequence number 1 on reverse strands
 None
----- prediction on sequence number 2 (length = 1040967, name = NW_020799093.1) -----

 Predicted genes for sequence number 2 on forward strands
 None
Predicted genes for sequence number 2 on reverse strands
 None
